### PR TITLE
[BEAM-1560] Use SimpleDoFnRunner to invoke and use new CombineFnRunner in batch mode of Flink runner

### PR DESCRIPTION
**Pull request by JingsongLi**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by JingsongLi** [comment]
R: @aljoscha

**Comment by aljoscha** [comment]
Run Flink RunnableOnService

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java [hunk 1]
```diff
@@ -640,7 +640,8 @@ public void translateNode(
               windowingStrategy,
               sideInputStrategies,
               context.getPipelineOptions(),
-              outputMap);
+              outputMap,
```
Comment (aljoscha) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java]: This file has three references to `OldFoFn` in comments. These should be changed to just `DoFn`.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkAggregatorFactory.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import org.apache.beam.runners.core.AggregatorFactory;
+import org.apache.beam.runners.core.ExecutionContext;
+import org.apache.beam.runners.flink.translation.wrappers.SerializableFnAggregatorWrapper;
+import org.apache.beam.sdk.transforms.Aggregator;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.flink.api.common.functions.RuntimeContext;
+
+/**
+ * A {@link AggregatorFactory} for the FlinkRunner.
```
Comment (aljoscha) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkAggregatorFactory.java]: This should probably say something like "for the Flink Batch Runner".

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkSideInputReader.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.sdk.util.SideInputReader;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.util.WindowingStrategy;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.flink.api.common.functions.RuntimeContext;
+
+/**
+ * A {@link SideInputReader} for the FlinkRunner.
```
Comment (aljoscha) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkSideInputReader.java]: This should be "for the Flink Batch Runner".

**Comment by JingsongLi** [comment]
Thank you for the nice review. Your title and description are more appropriate.

**Comment by aljoscha** [comment]
Just merged, thanks a lot for working on this! 😄

**Pull request closed without merging**